applications/callflow/src/cf_route_win.erl

```diff
@@ -237,14 +237,18 @@ store_owner_id(Call) ->
 -spec set_language(whapps_call:call()) -> whapps_call:call().
 set_language(Call) ->
     Default = wh_media_util:prompt_language(whapps_call:account_id(Call)),
-    case cf_endpoint:get(Call) of
-        {'ok', Endpoint} ->
-            Language = kz_device:language(Endpoint, Default),
-            lager:debug("setting language '~s' for this call", [Language]),
-            whapps_call:set_language(wh_util:to_lower_binary(Language), Call);
-        {'error', _E} ->
-            lager:debug("no source endpoint for this call, setting language to default ~s", [Default]),
-            whapps_call:set_language(Default, Call)
+    case whapps_call:language(Call) of
+        Default ->
+            case cf_endpoint:get(Call) of
+                {'ok', Endpoint} ->
+                    Language = kz_device:language(Endpoint, Default),
+                    lager:debug("setting language '~s' for this call", [Language]),
+                    whapps_call:set_language(wh_util:to_lower_binary(Language), Call);
+                {'error', _E} ->
+                    lager:debug("no source endpoint for this call, setting language to default ~s", [Default]),
+                    whapps_call:set_language(Default, Call)
```

Comment:
@danielfinke 
I don't think I understand this case clause.  If the default language is already set on the call and there is no source endpoint set the language of the call to the default (wouldn't the case clause ensure that already)?  

I suspect the intent is to only set the language if the language has not already been set but that doesn't seem like the safest means to make this determination.  It basically means if the language is set to something other than default it can not be changed during re-hunt, but if it is set to default (or just default) then it could be.  No?

Could you describe your use-case maybe it will clear things up or let me come up with some suggestions :)